lib/integrations/awsoidc/create_ec2ice.go

```diff
@@ -113,26 +130,35 @@ func CreateEC2ICE(ctx context.Context, clt CreateEC2ICEClient, req CreateEC2ICER
 		return nil, trace.Wrap(err)
 	}
 
-	createEC2ICEInput := &ec2.CreateInstanceConnectEndpointInput{
-		SubnetId:         &req.SubnetID,
-		SecurityGroupIds: req.SecurityGroupIDs,
-		TagSpecifications: []ec2types.TagSpecification{{
-			ResourceType: ec2types.ResourceTypeInstanceConnectEndpoint,
-			Tags:         req.ResourceCreationTags.ToEC2Tags(),
-		}},
-	}
-
-	ec2ICEndpoint, err := clt.CreateInstanceConnectEndpoint(ctx, createEC2ICEInput)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	endpointName := "unknown"
-	if ec2ICEndpoint.InstanceConnectEndpoint != nil {
-		endpointName = aws.ToString(ec2ICEndpoint.InstanceConnectEndpoint.InstanceConnectEndpointId)
+	createdEndpoints := make(map[string]string, len(req.Endpoints))
+	endpointNames := make([]string, 0, len(req.Endpoints))
+
+	for _, endpoint := range req.Endpoints {
+		createEC2ICEInput := &ec2.CreateInstanceConnectEndpointInput{
+			SubnetId:         &endpoint.SubnetID,
+			SecurityGroupIds: endpoint.SecurityGroupIDs,
+			TagSpecifications: []ec2types.TagSpecification{{
+				ResourceType: ec2types.ResourceTypeInstanceConnectEndpoint,
+				Tags:         req.ResourceCreationTags.ToEC2Tags(),
+			}},
+		}
+
+		ec2ICEndpoint, err := clt.CreateInstanceConnectEndpoint(ctx, createEC2ICEInput)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		endpointName := "unknown"
```

Comment:
This looks strange at a glance. Should the endpoint name be mandatory? Alternatively, should this behavior be documented?

Reply:
It's informational only
There are no API calls that use this.

AWS API has a lot of pointers. Usually we rely on `aws.ToString` to get its value or an empty string.

However, for nested values we have to do nil checks for the chain of fields.
In this case, we have to check for the value of `ec2ICEndpoint.InstanceConnectEndpoint`.

It should never be nil, but better be defensive IMO.

Instead of failing, returning a sentinel value seems a valid approach to be more resilient.

I'll add a comment stating this

Reply:
Thanks, a comment sure helps!